src/ConcertoEditor.tsx

```diff
@@ -0,0 +1,76 @@
+import * as monaco from '@monaco-editor/react';
+import { editor } from 'monaco-editor';
+
+const options:editor.IStandaloneEditorConstructionOptions = {
+  minimap: { enabled: false },
+  wordWrap: "on"
+}
+const concertoKeywords = ['concept','from','optional','default','range','regex','length','abstract','namespace','import', 'enum', 'scalar', 'extends', 'default', 'participant','asset', 'o','identified by','transaction','event'];
+const concertoTypes = ['String','Integer','Double','DateTime','Long','Boolean']
+
+export default function ConcertoEditor( {value, onChange} : {value: string, onChange?: (value:string|undefined) => void} ) {
+
+    function handleEditorWillMount(monaco:monaco.Monaco) {
+        monaco.languages.register({
+            id: 'concerto',
+            extensions: ['.cto'],
+            aliases: ['Concerto', 'concerto'],
+            mimetypes: ['application/concerto'],
+        });
+      
+        monaco.languages.setMonarchTokensProvider('concerto', {
```

Comment:
I'm sure this is good enough but have a look at the vscode extension for alignment on how we determine tokens https://github.com/accordproject/vscode-web-extension/blob/main/client/syntaxes/concerto.tmLanguage.json

just to align this editor with the official extension we publish.